### DISCUSSION TITLE
Not able to load nimBLE Arduino firmware to Adafruit NRF52 device

### DISCUSSION THREAD

**Discussion by andrevipid** [h2zero/NimBLE-Arduino#828] [Q&A] [answered]
Hello, as far as I noticed this NimBLE-Arduino library is available for both esp and nrf52:

![image](https://github.com/user-attachments/assets/6bf3963c-12aa-4cbe-b54e-25f6eeddfcea)

I have been trying to compile and load the code into Adafruit nRF52 dev board (https://www.adafruit.com/product/4481) however it seems not to be compatible as you can see:

I am using the example Server code:
https://github.com/h2zero/NimBLE-Arduino/blob/master/examples/NimBLE_Server/NimBLE_Server.ino

![image](https://github.com/user-attachments/assets/0248acbb-ffe3-401c-9188-a526b4dac6b3)

    
 However, I have got the following message in arduino ide output:
    
    
![image](https://github.com/user-attachments/assets/f8777a66-8c47-4f1b-8519-8122676e5abf)

Is this NimBLE compatible or not with nRF devices ?

**Comment by thorrak** [comment]
Are you using the n-able Arduino core available (here)[https://github.com/h2zero/n-able-Arduino]?

**Reply by andrevipid** [reply]
Hello @thorrak, yes I have added the following libraries/n-able folder:

![image](https://github.com/user-attachments/assets/0dd305af-5b55-4e5f-baef-be28ad8a32df)

Into the arduino ide sketch library:

![image](https://github.com/user-attachments/assets/ade41cc7-eec8-497e-becd-313e8acd98dd)

After adding, same erros occurs when compiling the firmware. The zip file that I have downloaded from github n-able contains several folders and they are not in the format required to be added as a library:

![image](https://github.com/user-attachments/assets/6c5421d0-f80e-46d5-a3cc-5bfbc04e8c07)

I saw in the read.me file that to use n-able library I should add that as a Board:

![image](https://github.com/user-attachments/assets/c48add11-db46-4eb9-ac76-d14e060ce10d)


However, in case I add as a board, how I will manage to use adafruit library with additional features ? Current board is the itsybitsy adafruit nrf52 board.

**Reply by thorrak** [reply, marked as answer]
Unfortunately, unless you follow the instructions for the n-able Arduino core and set it up as a board, you **aren't** using it, which explains why you aren't able to use NimBLE-Arduino. I recommend following the install instructions exactly in this case.

Treating this code as a library is not appropriate/how it is installed. 

I'm not sure what features you reference for the Adafruit library, but my guess is that they will work just fine. Try following the instructions for NimBLE-Arduino first and see if that solves your problem with this library, then work on debugging what might be broken with Adafruit's libraries.

**Reply by andrevipid** [reply]
> Unfortunately, unless you follow the instructions for the n-able Arduino core and set it up as a board, you **aren't** using it, which explains why you aren't able to use NimBLE-Arduino. I recommend following the install instructions exactly in this case.
> 
> Treating this code as a library is not appropriate/how it is installed.
> 
> I'm not sure what features you reference for the Adafruit library, but my guess is that they will work just fine. Try following the instructions for NimBLE-Arduino first and see if that solves your problem with this library, then work on debugging what might be broken with Adafruit's libraries.

Understood. I have done it however now:

![image](https://github.com/user-attachments/assets/ba05f262-c01c-497c-9aa2-efc84a8626c8)


However, errors still appears when compiling and uploading to the board:

![image](https://github.com/user-attachments/assets/846c9859-e882-4b28-ad14-6fbe61810af1)

Any guess ?

**Comment by andrevipid** [comment]
Thank you Sir, I managed to have it working following readme instructions on the library.
